FLASH_PAGE_SIZE on stm32h5

I've seen that this issue was patched with the new commit of hal_stm32, by removing the include of Legacy.
However, when I tried to build a connectedhomeip application, which rely on this KConfig, I still have the issue.
I don't have the issue with any other application. When I build, it seems that he is able to find Legacy

`/home/quentin/github/workspace/modules/hal/stm32/stm32cube/stm32h5xx/drivers/include/Legacy/stm32_hal_legacy.h:475:39: error: 'FLASH_PAGE_SIZE' undeclared (first use in this function); did you mean 'FLASH_BANK_SIZE'?`

If I comment the definition of `FLASH_PAGE`, everything works fine.

- platform : nucleo_h563zi
- patch : comment the definition

**To Reproduce**
Steps to reproduce the behavior:
1. add CONFIG_POSIX_API=y in prj.conf
2. west build
3. See error

**Expected behavior**
Application build with no zephyr related error


**Environment (please complete the following information):**
 - OS: Ubuntu 22.04 on WSL2
 - Toolchain : Zephyr SDK
 - Commit : main/master

@erwango Yes I have west updaye, and I am on the current main. I thought this problem was resolved by this previous PR https://github.com/zephyrproject-rtos/zephyr/pull/73363, which say that Legacy will not be included anymore. So why build system find it ? 

`/home/quentin/github/workspace/modules/hal/stm32/stm32cube/stm32h5xx/drivers/include/Legacy/stm32_hal_legacy.h:475:39: error: 'FLASH_PAGE_SIZE' undeclared (first use in this function); did you mean 'FLASH_BANK_SIZE'?`

--- a/west.yml
+++ b/west.yml
@@ -234,7 +234,7 @@ manifest:
       groups:
         - hal
     - name: hal_stm32
-      revision: 6443e5c4d391c3e9e283c5728cd07d855e663965
+      revision: 855f195793df094644ce8b8617c01275408bbf58
       path: modules/hal/stm32
       groups:
         - hal

